OsmAnd-java/src/main/java/net/osmand/binary/BinaryMapPoiReaderAdapter.java

```diff
@@ -452,16 +451,25 @@ private TIntLongHashMap readPoiNameIndex(Collator instance, String query, Search
 					}
 				}
 				if (listOfSepOffsets.size() > 0) {
-					offsets.putAll(listOfSepOffsets.get(0));
-					for (int j = 1; j < listOfSepOffsets.size(); j++) {
+					if (req.force) {
+						for (TIntLongHashMap m : listOfSepOffsets) {
+							int[] k = m.keys();
+							offsets.put(k[0], m.get(k[0]));
+						}
+					} else {
+						for (TIntLongHashMap m : listOfSepOffsets) {
+							offsets.putAll(m);
+						}
+					}
+					/*for (int j = 1; j < listOfSepOffsets.size(); j++) {
```

Comment:
better to not remove code I don't see any good replacement for offsets.retainAll(mp);